osu.Game/Rulesets/Mods/ModUsage.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Rulesets.Mods
+{
+    /// <summary>
+    /// The usage of this mod to determine whether it's playable in such context.
+    /// </summary>
+    public enum ModUsage
+    {
+        /// <summary>
+        /// Used for a per-user gameplay session. Determines whether the mod is playable by an end user.
+        /// </summary>
+        User,
+
+        /// <summary>
+        /// Used as a "required mod" for a multiplayer match.
+        /// </summary>
+        MultiplayerRequired,
+
+        /// <summary>
+        /// Used as a "free mod" for a multiplayer match.
+        /// </summary>
+        MultiplayerFree,
```

Comment:
This is still very confusing to me. Does `MultiplayerFree` imply that it can also be `MultiplayerRequired`? If so, this needs to be in the naming and documentation.

How about
```csharp
/// Mod can be used in multiplayer but must be applied to all users. This is generally the case for mods which affect the length of gameplay.
MultiplayerRoomWide,
// Mod can be used in multiplayer either at a room or per player level (ie. free mod selection).
MultiplayerPerPlayer,
```
focusing on the actual reasoning behind the limitations.

Reply:
Good suggestion, that looks to work well with the usages of this enum.